### PR TITLE
vim-patch:8.2.{3468,3617,3618,3622}: some other CWD related patches

### PR DESCRIPTION
**Pull request by zeertzjq**
#### vim-patch:8.2.3468: problem with :cd when editing file in non-existent directory

Problem:    Problem with :cd when editing file in non-existent directory. (Yee
            Cheng Chin)
Solution:   Prepend the current directory to get the full path.
https://github.com/vim/vim/commit/c6376c798433bcb9ee38a8664299d11454546950

#### vim-patch:8.2.3617: ":verbose pwd" does not mention 'autochdir' was applied

Problem:    ":verbose pwd" does not mention 'autochdir' was applied.
Solution:   Remember the last chdir was done by 'autochdir'.  (issue vim/vim#9142)
https://github.com/vim/vim/commit/0526815c15170a5926e1008600ec29d42d8b64c2

#### vim-patch:8.2.3618: getcwd() is unclear about how 'autochdir' is used

Problem:    getcwd() is unclear about how 'autochdir' is used.
Solution:   Update the help for getcwd().  Without any arguments always return
            the actual current directory.
https://github.com/vim/vim/commit/851c7a699ae00bdc14a4db874cf722b7b7393b53

#### vim-patch:8.2.3622: "verbose pwd" shows confusing info

Problem:    "verbose pwd" shows confusing info when :lcd does not change
            directory.
Solution:   Clear last_chdir_reason also when the directory does not change.
            (closes vim/vim#9160)
https://github.com/vim/vim/commit/64be6aa3a54ecfe355d4a03e1200650c301e7f29

### CODE REVIEW

**Comment by zeertzjq** [comment]
The change from `zipfile:` to `zipfile://` was introduced to Vim in <https://github.com/vim/vim/commit/519cc559b08b800edc429688aece7ad6a00d41eb#diff-c3431e35dce9ef0098df28d227c4539fc744a55e32dca0291be3eeddab0fe245> and ported to Nvim in <https://github.com/neovim/neovim/commit/c0efe49e78eca3a17a5934cc283199122d081467#diff-c3431e35dce9ef0098df28d227c4539fc744a55e32dca0291be3eeddab0fe245>.

**Comment by zeertzjq** [comment]
Vim has a solution to #9892 with fewer side-effects in patch 8.2.3618

**Comment by zeertzjq** [comment]
I'll wait for <https://github.com/vim/vim/pull/9160>.

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -3535,12 +3530,13 @@ static void f_getcwd(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
-    } else if (os_dirname(cwd, MAXPATHL) == FAIL) {  // Get the OS CWD.
+      break;
+    }
+    FALLTHROUGH;            // In global directory, just need to get OS CWD.
+  case kCdScopeInvalid:     // If called without any arguments, get OS CWD.
+    if (os_dirname(cwd, MAXPATHL) == FAIL) {
       from = (char_u *)"";  // Return empty string on failure.
     }
-    break;
-  case kCdScopeInvalid:     // We should never get here
```
Comment (janlazo) [inline src/nvim/eval/funcs.c:3542]: Does the static analyzer report that this is unreachable?

Comment (zeertzjq) [reply src/nvim/eval/funcs.c:3542]: I also deleted lines 3488-3492

File: src/nvim/eval/funcs.c [hunk 2]
```diff
@@ -3535,12 +3530,13 @@ static void f_getcwd(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
       from = globaldir;
-    } else if (os_dirname(cwd, MAXPATHL) == FAIL) {  // Get the OS CWD.
+      break;
+    }
+    FALLTHROUGH;            // In global directory, just need to get OS CWD.
+  case kCdScopeInvalid:     // If called without any arguments, get OS CWD.
+    if (os_dirname(cwd, MAXPATHL) == FAIL) {
       from = (char_u *)"";  // Return empty string on failure.
     }
-    break;
```
Comment (janlazo) [inline src/nvim/eval/funcs.c:3541]: implicit fallthrough even though it's the last case.

**Pull request merged**